tests/resources/test_collection.py

```diff
@@ -303,3 +303,133 @@ async def test_get_collections_search(
+    # assert {"root", "self"} == {link["rel"] for link in links}
+
+    ###################
+    # limit=3, there should not be a next/previous link
+    resp = await app_client.get(
+        "/collections",
+        params={"limit": 3},
+    )
+    cols = resp.json()["collections"]
+    links = resp.json()["links"]
+    assert len(cols) == 2
+    assert cols[0]["id"] == load_test_collection["id"]
+    assert cols[1]["id"] == load_test2_collection.id
+    assert len(links) == 2
+    assert {"root", "self"} == {link["rel"] for link in links}
+
+    ###################
+    # offset=3, because there are 2 collections, we should not have `next` or `prev` links
+    resp = await app_client.get(
+        "/collections",
+        params={"offset": 3},
+    )
+    cols = resp.json()["collections"]
+    links = resp.json()["links"]
+    assert len(cols) == 0
+    assert len(links) == 2
+    assert {"root", "self"} == {link["rel"] for link in links}
+
+    ###################
+    # offset=3,limit=1
+    resp = await app_client.get(
+        "/collections",
+        params={"limit": 1, "offset": 3},
+    )
+    cols = resp.json()["collections"]
+    links = resp.json()["links"]
+    assert len(cols) == 0
+    assert len(links) == 3
+    assert {"root", "self", "previous"} == {link["rel"] for link in links}
+    prev_link = list(filter(lambda link: link["rel"] == "previous", links))[0]
+    assert prev_link["href"].endswith("?limit=1&offset=2")
+
+    # ###################
+    # # offset=3,limit=2
+    # resp = await app_client.get(
+    #     "/collections",
+    #     params={"limit": 2,"offset": 3},
+    # )
+    # cols = resp.json()["collections"]
+    # links = resp.json()["links"]
+    # assert len(cols) == 0
+    # assert len(links) == 3
+    # assert {"root", "self", "previous"} == {link["rel"] for link in links}
+    # prev_link = list(filter(lambda link: link["rel"] == "previous", links))[0]
+    # assert prev_link["href"].endswith("?limit=1&offset=2")
+
+    ###################
+    # offset=1, should have a `previous` link
+    resp = await app_client.get(
+        "/collections",
+        params={"offset": 1},
+    )
+    cols = resp.json()["collections"]
+    links = resp.json()["links"]
+    assert len(cols) == 1
+    assert cols[0]["id"] == load_test2_collection.id
+    # TODO: Check with pgstac
+    # assert len(links) == 3
+    # assert {"root", "self", "previous"} == {link["rel"] for link in links}
+    # prev_link = list(filter(lambda link: link["rel"] == "previous", links))[0]
+    # assert prev_link["href"].endswith("?offset=0")
+
+    ###################
+    # offset=0, should not have next/previous link
+    resp = await app_client.get(
+        "/collections",
+        params={"offset": 0},
+    )
+    cols = resp.json()["collections"]
+    links = resp.json()["links"]
+    assert len(cols) == 2
+    # TODO: Check with pgstac
+    # assert len(links) == 2
+    # assert {"root", "self"} == {link["rel"] for link in links}
```

Comment:
there seems to be bugs in pgstac about how next/previous links are created

should open an issue in pgstac repo